### PR TITLE
secvarctl-tests.py: Add secvarctl test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add secvarctl test cases.
This package helps in creating keys.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 7c41e9580e32a01e6434b5d75152a821eb769e76
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-10T00.05-7c41e95/job.log
 (1/2) secvarctl-tests.py:secvarctl.test;run-run_type-upstream-131e: PASS (11.45 s)
 (2/2) secvarctl-tests.py:secvarctl.test;run-run_type-distro-17a2: ERROR: secvarctl source install failed. (1.78 s)
RESULTS    : PASS 1 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-10T00.05-7c41e95/results.html
JOB TIME   : 25.85 s

SLES:
JOB ID     : fd4c63bd10e570606805c486bb8de44abbc805c1
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-10T10.39-fd4c63b/job.log
 (1/2) secvarctl-tests.py:secvarctl.test;run-run_type-upstream-131e: PASS (10.04 s)
 (2/2) secvarctl-tests.py:secvarctl.test;run-run_type-distro-17a2: PASS (13.44 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-10T10.39-fd4c63b/results.html
JOB TIME   : 44.79 s
```

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : f533df389e827ca6fc6288a9abfccf8af0d1038b
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-10T00.31-f533df3/job.log
 (1/2) secvarctl-tests.py:secvarctl.test;run-run_type-upstream-131e: PASS (11.26 s)
 (2/2) secvarctl-tests.py:secvarctl.test;run-run_type-distro-17a2: ERROR: secvarctl source install failed. (1.72 s)
RESULTS    : PASS 1 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-10T00.31-f533df3/results.html
JOB TIME   : 25.34 s

SLES:
JOB ID     : af0f04e3167d40357f6bdf8b1860306302bf77b9
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-10T10.59-af0f04e/job.log
 (1/2) secvarctl-tests.py:secvarctl.test;run-run_type-upstream-131e: PASS (12.27 s)
 (2/2) secvarctl-tests.py:secvarctl.test;run-run_type-distro-17a2: PASS (13.71 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-10T10.59-af0f04e/results.html
JOB TIME   : 47.34 s
```

File: security/secvarctl-tests.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        if output.exit_status:
+            self.fail("secvarctl 'make check' failed.")
+        # TBD the following
+        for line in output.stdout_text.splitlines():
+            if 'FAIL:' in line and 'XFAIL:' not in line and \
+               '# FAIL:' not in line:
+                count += 1
+                self.log.info(line)
+        if count:
+            self.fail("%s test(s) failed, please refer to the log" % count)
```
Comment (Naresh-ibm) [inline security/secvarctl-tests.py:72]: are we not running any test here? I could see only compiling the source code here.
please let me know if I am missing something here.

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Comment by Naresh-ibm** [comment]
@nasastry please add a README file which will help new user to get the glims of the test.
having the debug.log will be more helpfull for new test cases.

File: security/secvarctl-tests.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    def test(self):
+        '''
+        Running tests from secvarctl
+        '''
+        count = 0
+        output = build.run_make(self.srcdir, extra_args="OPENSSL=1",
+                                process_kwargs={"ignore_status": True})
+        if output.exit_status:
+            self.fail("secvarctl 'make check' failed.")
+        # TBD the following
```
Comment (abdhaleegit) [inline security/secvarctl-tests.py:65]: any thing TBD here, or we can remove that line ?

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: security/secvarctl-tests.py (hunk 1 above)
Comment (abdhaleegit) [reply security/secvarctl-tests.py:72]: @Naresh-ibm basically make check will start all the test

**Review by abdhaleegit — APPROVED** [review]
Apart from what test does information rest all LGTM

**Comment by nasastry** [comment]
Added 'README.txt' and attaching the debug/job.log files.
[rhel_secvarctl_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9817190/rhel_secvarctl_job.log)
[sles_secvarctl_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9817194/sles_secvarctl_job.log)

**Review by Naresh-ibm — APPROVED** [review]
LGTM

**Comment by Naresh-ibm** [comment]
Thank you @nasastry for the PR and @abdhaleegit for reviews

**Pull request merged**